server/controllers/UserController.js

```diff
@@ -2,73 +2,90 @@ const passport = require('passport');
 const UserService = require('../services/UserService');
 
 // Checks if it's a guest or user entering webpage
-exports.main = (req, res) => {
+const main = (req, res) => {
   UserService.getSession(req.session.token).then((data) => {
     req.session.token = data.token;
     res.json(data.user);
   });
 };
 
-exports.getUser = (req, res) => {
+const getUser = (req, res) => {
   UserService.getUser(req.params.userID).then((records) => {
     res.json(records);
   });
 };
 
-exports.getUserToDoLabs = (req, res) => {
+const getUserToDoLabs = (req, res) => {
   UserService.getUserToDoLabs(req.params.userID).then((records) => {
     res.json(records);
   });
 };
-exports.getUserAssignedLabs = (req, res) => {
+const getUserAssignedLabs = (req, res) => {
   UserService.getUserAssignedLabs(req.params.userID).then((records) => {
     res.json(records);
   });
 };
 
-exports.getUserEnrolledGroups = (req, res) => {
+const getUserEnrolledGroups = (req, res) => {
   UserService.getUserEnrolledGroups(req.params.userID).then((records) => {
     res.json(records);
   });
 };
 
-exports.getUserInstructingGroups = (req, res) => {
+const getUserInstructingGroups = (req, res) => {
   UserService.getUserInstructingGroups(req.params.userID).then((records) => {
     res.json(records);
   });
 };
 
 // Authenticates User through Google OAuth
-exports.authenticate = passport.authenticate('google', {
+const authenticate = passport.authenticate('google', {
   scope: ['email', 'profile'],
 });
 
 // Callback used for Google OAuth
-exports.authenticateRedirect = passport.authenticate('google', {
+const authenticateRedirect = passport.authenticate('google', {
   keepSessionInfo: true,
   failureRedirect: '/',
 });
 
-exports.authenticateCallback = (req, res) => {
-  UserService.authenticate(req.user.profile).then((data) => {
-    UserService.updateGuestUserId(data.userid, req.session.token).then(()=>{
-      req.session.token = data.usersessionid;
-      res.redirect(req.session.url);
-    },
-    );
-  });
+const authenticateCallback = async (req, res) => {
+  try {
+    const data = await UserService.authenticate(req.user.profile);
+    req.session.token = data.usersessionid;
+    await UserService.updateGuestUserId(data.userid, req.session.token);
+    res.status(301);
```

Comment:
Why 301?

Reply:
HTTP 301 status indicates permanent redirect, and will update the current URL to the desired redirect location (whatever page the user was one when they logged in)

Reply:
I see. I didn't see the `res.redirect(req.session.url)`. nice